doc/sphinx/language/core/assumptions.rst

```diff
@@ -90,7 +90,9 @@ These terms are also useful:
 
 * `n : nat` is a :gdef:`dependent premise` of `forall n:nat, n + 0 = n` because
   `n` appears both in the binder of the `forall` and in the quantified statement
-  `n + 0 = n`.
+  `n + 0 = n`.  Note that if `n` doesn't appear in the statement, Coq considers it
+  a non-dependent premise.  Similarly, :n:`let n := ... in @term` is a
+  dependent premise only if `n` is used in :n:`@term`.
 
```

Comment:
Perhaps we can be more clear. It might be that `n` doesn't "appear" in the statement since it is an implicit argument to the statement. We would need to have `Set Printing All` to be sure. What we could say instead is: "if `n` doesn't occur in the statement".

Reply:
I'll go with "isn't used"